op-deployer/pkg/deployer/pipeline/l2genesis.go

```diff
@@ -93,9 +91,10 @@ func GenerateL2Genesis(pEnv *Env, intent *state.Intent, bundle ArtifactsBundle,
 		SequencerFeeVaultRecipient:               thisIntent.SequencerFeeVaultRecipient,
 		GovernanceTokenOwner:                     overrides.GovernanceTokenOwner,
 		Fork:                                     big.NewInt(schedule.SolidityForkNumber(1)),
-		UseInterop:                               intent.UseInterop,
-		EnableGovernance:                         overrides.EnableGovernance,
-		FundDevAccounts:                          overrides.FundDevAccounts,
+		// Only include interop predeploys if it is activating at genesis
+		UseInterop:       schedule.L2GenesisInteropTimeOffset != nil && *schedule.L2GenesisInteropTimeOffset == 0,
```

Comment:
The `L2Genesis.s.sol` script depends on `useInterop` here still: https://github.com/ethereum-optimism/optimism/blob/12c5398a1725a2aafc3e7abb0711cf761a2b20b1/packages/contracts-bedrock/scripts/L2Genesis.s.sol#L194

We should fix the `L2Genesis` script before we can remove the `useInterop` field from the `UpgradeScheduleDeployConfig` type safely.

This line here thus makes sense to keep `useInterop` around with.
I'll create a github issue to track the removal of `useInterop` from the `L2Genesis` script.

Reply:
Tracking here: https://github.com/ethereum-optimism/optimism/issues/16075